src/pulse3D/plate_recording.py

```diff
@@ -557,10 +557,14 @@ def to_dataframe(self) -> pd.DataFrame:
 
         data = {"Time (s)": pd.Series(interp_timepoints)}
 
-        output_stim_data = not self.is_optical_recording and first_well.version >= VersionInfo.parse("1.0.0")
+        output_stim_data = (
+            not self.is_optical_recording
+            and first_well.version >= VersionInfo.parse(MIN_FILE_VERSION_FOR_STIM_INTERPOLATION)
+            or include_stim_data
+        )
 
         # add stim timepoints
-        if output_stim_data:
+        if output_stim_data or include_stim_data:
```

Comment:
This still isn't what we want. It should just be `if output_stim_data:`. The way it's set up now would try to output stim data from an optical file still because of the `or include_stim_data`. This is what I meant when I mentioned we need to add tests for the various cases, we need to test different types of files as well